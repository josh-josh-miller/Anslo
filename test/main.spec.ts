import Anslo from '../src'

class User {
  self: User
  listOfNumbers: number[] = [1, 2, 3, 4, 5, 6]
  created: Date = new Date("2022-10-10")
}

describe("anslo", () => {
  it("can blow on bad model", () => {
    expect(() => new Anslo(null)).toThrowErrorMatchingSnapshot()
  })

  it("can serialize and deserialize", () => {
    const user = new User()
    user.self = user
    const anslo = new Anslo({
      User,
    })
    const serialization = anslo.down(user)
    const userTwo = anslo.up(serialization)
    expect(user).toEqual(userTwo)
    expect(serialization).toMatchSnapshot()
  })

  it("fails to parse an invalid string", () => {
    const anslo = new Anslo({})
    expect(() => anslo.up("{")).toThrowErrorMatchingSnapshot()
  })

  it(`can serialize with spaces`, () => {
    const user = new User()
    user.self = user
    const anslo = new Anslo({
      User,
    })
    expect(anslo.down(user, 4)).toMatchSnapshot()
  })
})
