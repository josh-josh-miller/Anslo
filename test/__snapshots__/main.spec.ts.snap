// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anslo can blow on bad model 1`] = `"On construction, new Anslo( ===> models <===) was not an object of constructors."`;

exports[`anslo can serialize and deserialize 1`] = `"{\\"pointers\\":[{},[],1,2,3,4,5,6,{}],\\"graph\\":{\\"r\\":0,\\"t\\":\\"object\\",\\"c\\":{\\"listOfNumbers\\":{\\"r\\":1,\\"t\\":\\"array\\",\\"c\\":{\\"0\\":{\\"r\\":2,\\"t\\":\\"number\\"},\\"1\\":{\\"r\\":3,\\"t\\":\\"number\\"},\\"2\\":{\\"r\\":4,\\"t\\":\\"number\\"},\\"3\\":{\\"r\\":5,\\"t\\":\\"number\\"},\\"4\\":{\\"r\\":6,\\"t\\":\\"number\\"},\\"5\\":{\\"r\\":7,\\"t\\":\\"number\\"}}},\\"created\\":{\\"r\\":8,\\"t\\":\\"date\\",\\"v\\":\\"2022-10-10T00:00:00.000Z\\"},\\"self\\":{\\"r\\":0,\\"t\\":\\"object\\",\\"cx\\":\\"User\\"}},\\"cx\\":\\"User\\"}}"`;

exports[`anslo can serialize with spaces 1`] = `
"{
    \\"pointers\\": [
        {},
        [],
        1,
        2,
        3,
        4,
        5,
        6,
        {}
    ],
    \\"graph\\": {
        \\"r\\": 0,
        \\"t\\": \\"object\\",
        \\"c\\": {
            \\"listOfNumbers\\": {
                \\"r\\": 1,
                \\"t\\": \\"array\\",
                \\"c\\": {
                    \\"0\\": {
                        \\"r\\": 2,
                        \\"t\\": \\"number\\"
                    },
                    \\"1\\": {
                        \\"r\\": 3,
                        \\"t\\": \\"number\\"
                    },
                    \\"2\\": {
                        \\"r\\": 4,
                        \\"t\\": \\"number\\"
                    },
                    \\"3\\": {
                        \\"r\\": 5,
                        \\"t\\": \\"number\\"
                    },
                    \\"4\\": {
                        \\"r\\": 6,
                        \\"t\\": \\"number\\"
                    },
                    \\"5\\": {
                        \\"r\\": 7,
                        \\"t\\": \\"number\\"
                    }
                }
            },
            \\"created\\": {
                \\"r\\": 8,
                \\"t\\": \\"date\\",
                \\"v\\": \\"2022-10-10T00:00:00.000Z\\"
            },
            \\"self\\": {
                \\"r\\": 0,
                \\"t\\": \\"object\\",
                \\"cx\\": \\"User\\"
            }
        },
        \\"cx\\": \\"User\\"
    }
}"
`;

exports[`anslo fails to parse an invalid string 1`] = `"Could not parse string."`;
